src/s3/commands/createFolder.ts

```diff
@@ -0,0 +1,85 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DEFAULT_DELIMITER } from '../../shared/clients/s3Client'
+import { getLogger } from '../../shared/logger'
+import * as telemetry from '../../shared/telemetry/telemetry'
+import { S3BucketNode } from '../explorer/s3BucketNode'
+import { S3FolderNode } from '../explorer/s3FolderNode'
+import { Commands, DefaultCommands } from '../../shared/vscode/commands'
+import { DefaultWindow, Window } from '../../shared/vscode/window'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+/**
+ * Creates a subfolder in the bucket or folder represented by the given node.
+ *
+ * Prompts the user for the folder name, applying basic validations.
+ * Creates the folder.
+ * Refreshes the node.
+ *
+ * Node that the node is reset to displaying its first page of results.
+ * The folder that is created won't necessary fall on the first page.
+ * The user may need to load more pages to see the created folder reflected in the tree.
```

Comment:
Is this what a user would expect?

Reply:
It's subtle but it's the only way to ensure the ordering is stable across refreshes and it's quite a bit easier to implement. This is also the behavior of the S3 console.

The alternative would be to stick the new folder at the bottom of the current page. Then it would be in a completely different page / order upon refresh. You also have to deal with filtering the entry out at a later time if the user clicks "load more".

I favor this for its simplicity.

Reply:
:+1: for matching with the console. That being said, could you add a toast for confirmation?

Reply:
That's a good idea for more visual feedback. 

The upload/download show the progress bar notification so I didn't add a notification for those. Do you think it'd be useful to show a "`Download[/Upload] complete`" notification in addition to the progress for those?

Reply:
Yes for folders since they'll complete so fast that there won't be any time to see the bar. Might be nice for items, although it could get annoying if the user is trying to upload/download a lot of things (we won't support a bulk upload/download, right? That would make it less annoying)

Reply:
Added a confirmation message for both `Create Folder` and `Create Bucket`